File: periods.py
import numpy as np
from parameters import MIN_ID, MIN_D, INF_RED, AQ, AG


def id_time(base_period, infection_count):
    return np.round(
        (base_period - MIN_ID)
        * np.exp(-INF_RED * (infection_count - 1))
        + MIN_ID
    )


def d_time(base_period, infection_count, ages):
    return np.round(
        (base_period - MIN_D)
        * np.exp(AQ * (1 - infection_count) - AG * ages)
        + MIN_D
        )

def latent_time(base_period):
    return base_period
        

File: periods.c
#include <math.h>

#define MIN_ID 11
#define MIN_D 1
#define IND_RED 0.45
#define AQ 0.00179
#define AG 0.0368

int setidtime(double base, int count, int age) {
  double b = base - MIN_ID;
  double a = INF_RED * (1 - count);
  return (int)floor(b * exp(a) + MIN_ID);
}

int setdtime(double base, int count, int age) {
  double b = base - MIN_D;
  double a = AQ * (1 - count) - AG * age;
  return (int)floor(b * exp(a) + MIN_D);
}

int setlatenttime(double base, int count, int age) {
  return base;
}
